clustering/kubernetes.py

```diff
@@ -0,0 +1,449 @@
+            state=dict(default="present",
+                       choices=["present", "post",
+                                "absent", "delete",
+                                "update", "put",
+                                "replace", "patch",
+                                "kubectl"])
+        )
+    )
+
+    decode_cert_data(module)
+
+    changed = False
+    data = module.params.get('inline_data', {})
+    if not data:
+        dfile = module.params.get('file_reference')
+        if dfile:
+            f = open(dfile, "r")
+            data = yaml.load(f)
+
+    endpoint = "https://" + module.params.get('api_endpoint')
+    url = endpoint
+
+    namespace = "default"
+    if data and 'metadata' in data:
+        namespace = data['metadata'].get('namespace', "default")
+        kind = data['kind'].lower()
+        url = endpoint + KIND_URL[kind]
+        url = url.replace("{namespace}", namespace)
+
+    # check for 'kubectl' commands
+    kubectl_api_versions = module.params.get('kubectl_api_versions')
+    kubectl_cluster_info = module.params.get('kubectl_cluster_info')
+    kubectl_get = module.params.get('kubectl_get')
+    kubectl_namespace = module.params.get('kubectl_namespace')
+    kubectl_version = module.params.get('kubectl_version')
+
+    state = module.params.get('state')
+    if state in ['present', 'post']:
+        changed, body = k8s_create_resource(module, url, data)
+        module.exit_json(changed=changed, api_response=body)
+
+    if state in ['absent', 'delete']:
+        changed, body = k8s_delete_resource(module, url, data)
+        module.exit_json(changed=changed, api_response=body)
+
+    if state in ['replace', 'put']:
+        changed, body = k8s_replace_resource(module, url, data)
+        module.exit_json(changed=changed, api_response=body)
+
+    if state in ['update', 'patch']:
+        changed, body = k8s_update_resource(module, url, data)
+        module.exit_json(changed=changed, api_response=body)
+
+    if state == 'kubectl':
+        kurl = url + "/api/v1/namespaces/" + kubectl_namespace
+        if kubectl_get:
+            if kubectl_get.startswith("namespaces"):
+                kurl = url + "/api/v1"
+            changed, body = k8s_kubectl_get(module, kurl)
+            module.exit_json(changed=changed, api_response=body)
+        if kubectl_version:
+            info, body = api_request(module, url + "/version")
+            module.exit_json(changed=False, api_response=body)
+        if kubectl_api_versions:
+            info, body = api_request(module, url + "/api")
+            module.exit_json(changed=False, api_response=body)
+        if kubectl_cluster_info:
+            info, body = api_request(module, url +
+                                     "/api/v1/namespaces/kube-system"
+                                     "/services?labelSelector=kubernetes"
+                                     ".io/cluster-service=true")
+            module.exit_json(changed=False, api_response=body)
```

Comment:
I would expect these to exist in a 'facts' module instead of being part of this module.